### PR TITLE
Resource manager cleanup

### PR DESCRIPTION
**Pull request by talekhinezh**

### CODE REVIEW

File: radix-engine/src/engine/track.rs [hunk 1]
```diff
@@ -456,31 +461,51 @@ impl<'s, S: SubstateStore> Track<'s, S> {
...
                 SubstateUpdate {
-                    prev_id: substate_update.prev_id,
+                    prev_id,
                     value: resource_manager,
                 },
             );
-            Ok(result)
         } else {
-            panic!("Resource Manager not found");
+            panic!("Component was never borrowed");
```
Comment (iamyulong) [inline radix-engine/src/engine/track.rs:503]: Message is misleading :)

File: radix-engine/src/engine/track.rs [hunk 2]
```diff
@@ -456,31 +461,51 @@ impl<'s, S: SubstateStore> Track<'s, S> {
...
+        let maybe_resource = self.resource_managers.remove(&resource_address);
+        if let Some(SubstateUpdate { value, prev_id }) = maybe_resource {
+            self.borrowed_resource_managers
+                .insert(resource_address, prev_id);
+            Ok(value)
+        } else if self
+            .borrowed_resource_managers
+            .contains_key(&resource_address)
+        {
+            panic!("Invalid resource manager reentrancy");
```
Comment (iamyulong) [inline radix-engine/src/engine/track.rs:477]: Should this be an error instead of panic? Would also consider moving it up for readability.

Comment (iamyulong) [inline radix-engine/src/engine/track.rs:477]: Should this be an error instead of panic?

Comment (talekhinezh) [reply radix-engine/src/engine/track.rs:477]: No, panic for now since this logic is all enforced by system at the moment

**Review by iamyulong — APPROVED** [review]

**Pull request merged**